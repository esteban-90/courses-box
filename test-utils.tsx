import { FC, ReactElement, ReactNode } from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ThemeProvider } from '@emotion/react'
import { Themes } from '@/styles/themes'

const Wrapper: FC<{ children: ReactNode }> = ({ children }): JSX.Element => {
  return <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: RenderOptions): RenderResult => {
  return render(ui, { wrapper: Wrapper, ...options })
}

export * from '@testing-library/react'
export { customRender as render }

// 14
