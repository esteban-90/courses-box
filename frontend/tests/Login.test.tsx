import userEvent from '@testing-library/user-event'
import { render, screen, act } from '@/test-utils'
import Login from '@/pages/login'

describe('Login Page test cases', () => {
  it('Render check', () => {
    const { container } = render(<Login />)
    const loginForm = screen.getByTestId('loginForm')

    expect(container).toMatchSnapshot()
    expect(loginForm).toHaveFormValues({ identifier: '', password: '' })
  })

  it('Client validation check', async () => {
    render(<Login />)

    const submitButton = screen.getByRole('button', { name: 'Enter' })
    await act(async () => await userEvent.click(submitButton))

    const requiredIdentifierFeedback = screen.getByText('Identifier is required')
    const requiredPasswordFeedback = screen.getByText('Password is required')

    expect(requiredIdentifierFeedback).toBeInTheDocument()
    expect(requiredPasswordFeedback).toBeInTheDocument()

    const identifierField = screen.getByTestId('loginIdentifier')
    const passwordField = screen.getByTestId('loginPassword')

    await act(async () => {
      await userEvent.type(identifierField, 'test@example.com')
      await userEvent.type(passwordField, 'Test1234*')
    })

    const validFeedbacks = screen.getAllByRole('alert')
    const [identifierValidFeedback, passwordValidFeedback] = validFeedbacks

    expect(validFeedbacks).toHaveLength(2)
    expect(identifierValidFeedback).toMatchSnapshot()
    expect(passwordValidFeedback).toMatchSnapshot()
  })
})
