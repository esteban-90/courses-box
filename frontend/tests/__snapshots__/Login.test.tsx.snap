// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page test cases Client validation check 1`] = `
<span
  class="css-1reg724"
  role="alert"
>
   
</span>
`;

exports[`Login Page test cases Client validation check 2`] = `
<span
  class="css-1reg724"
  role="alert"
>
   
</span>
`;

exports[`Login Page test cases Render check 1`] = `
<div>
  <form
    data-testid="loginForm"
    novalidate=""
  >
    <div
      class="css-gbwc6v"
    >
      <section
        class="css-1c469nw"
      >
        <h2>
          Login Page
        </h2>
        <label
          class="css-fjegpl"
          height="7"
          width="20"
        >
          <span
            class="css-1grri41"
          >
            Your username or email:
          </span>
          <div
            class="css-11gjszk"
          >
            <input
              class="css-1h6wv30"
              data-testid="loginIdentifier"
              name="identifier"
              placeholder="..."
            />
          </div>
          <span
            class="css-1reg724"
            role="alert"
          >
             
          </span>
        </label>
        <label
          class="css-fjegpl"
          height="7"
          width="20"
        >
          <span
            class="css-1grri41"
          >
            Your password:
          </span>
          <div
            class="css-11gjszk"
          >
            <input
              class="css-1h6wv30"
              data-testid="loginPassword"
              maxlength="10"
              minlength="8"
              name="password"
              placeholder="**********"
              type="password"
            />
          </div>
          <span
            class="css-1reg724"
            role="alert"
          >
             
          </span>
        </label>
        <button
          class="css-tp74ot"
          style="margin-top: 1rem;"
          type="submit"
        >
          Enter
        </button>
        <a
          class="css-1ghpz66"
          href="/registration"
          style="margin-top: 1rem; font-weight: bold;"
        >
          Create your account
        </a>
      </section>
    </div>
  </form>
</div>
`;
